### PR TITLE
Fixes #17483: Validate platform input data before running a scenario on it

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/17483

### CODE REVIEW

File: rudder-tests/data.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (peckpeck) [inline rudder-tests/data.json:1]: what is this file at the root of the directory and how is it generated ?

File: rudder-tests/schema.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import re
+import sys
+import traceback
+from subprocess import Popen, PIPE
+from jsonschema import validate, draft7_format_checker, Draft7Validator, RefResolver
+import importlib
+
+###################
+
+def run_scenario(name, datastate, json_file=None):
```
Comment (peckpeck) [inline rudder-tests/schema.py:17]: I don't understand, this file is called schema.py and contains a function to run a scenario ?

**Comment by Fdall** [comment]
PR updated with a new commit

File: rudder-tests/spec/spec_helper.rb [hunk 3]
```diff
@@ -53,6 +54,6 @@
 
 
 RSpec.configure do |c|
-  c.output_stream = File.open('serverspec-result.xml', 'w')
+  c.output_stream = File.open(workspace + '/serverspec-result.xml', 'w')
```
Comment (peckpeck) [inline rudder-tests/spec/spec_helper.rb:57]: if workspace is empty this will write to /
can't you put the / at the end of the variable instead ?

**Comment by Fdall** [comment]
PR updated with a new commit

**Pull request merged**